Build guide docs - build docs from local (#789)
PowerGridModel/power-grid-model: docs/advanced_documentation/build-guide.md

This project has the main project as a required dependency. Configuration will fail if the main project has not been
built and installed, e.g. using `cmake --build --preset <preset> --target install` for the current preset.
```

## Documentation

> The documentation is built in [Sphinx](https://github.com/sphinx-doc/sphinx). It can be built locally in python environment. The packages required for building it can be found under [docs]. The steps required for building the python package needs to be carried out first as per (#build-python-package). Then the documentation specific packages can be installed via:

Reviewer: ```suggestion
The documentation is built in [Sphinx](https://github.com/sphinx-doc/sphinx). It can be built locally in a Python environment. The packages required for building it can be found under the `[docs]` optional dependencies. In addition, the `power-grid-model` Python package needs to be built by following the steps mentioned in the [above](#build-python-package). After that, the documentation specific packages can be installed via:

```

```shell

> pip install -e .[doc]

Reviewer: ```suggestion
pip install -e .[docs]
```

Author: Its under [doc]

Reviewer: in that case, please update the `[docs]` in the above section

> pip install -e .[doc]

Reviewer: this will again install `.` just like the prebuild steps in #build-python-package. that may cause confusion for the reader. please rephrase the section above

Author: I dont think theres a way to install optional dependencies without installing `.`

Reviewer: no it's about rephrasing

```

> [Doxygen](https://www.doxygen.nl/) needs to be installed to generate C API documentation. (Building C API documentation can be manually disabled by commenting out `breathe` settings in  `docs/conf.py` if required.)

Reviewer: ```suggestion

The C API documentation is generated using [Doxygen](https://www.doxygen.nl). If you do not have Doxygen installed, it can also be temporarily bypassed by commenting out the `breathe` settings in  `docs/conf.py`.
```

This project has the main project as a required dependency. Configuration will fail if the main project has not been
built and installed, e.g. using `cmake --build --preset <preset> --target install` for the current preset.
```

## Documentation

> The documentation is built in [Sphinx](https://github.com/sphinx-doc/sphinx). It can be built locally in a Python environment. The packages required for building it can be found under the `[docs]` optional dependencies. In addition, the `power-grid-model` Python package needs to be built by following the steps mentioned in the [above](#build-python-package). After that, the documentation specific packages can be installed via:

Reviewer 2: ```suggestion
The documentation is built in [Sphinx](https://github.com/sphinx-doc/sphinx). It can be built locally in a Python environment. The packages required for building it can be found under the `[docs]` optional dependencies. In addition, the `power-grid-model` Python package needs to be built by following the steps mentioned [above](#build-python-package). After that, the documentation specific packages can be installed via:
```

```shell
pip install -e .[doc]
```

The C API documentation is generated using [Doxygen](https://www.doxygen.nl). If you do not have Doxygen installed, it can also be temporarily bypassed by commenting out the `breathe` settings in  `docs/conf.py`.

> The documentation can be built by following which gives out the html in `docs/_build/html` directory.

Reviewer 2: ```suggestion
The documentation can be built with the following commands, resulting in an html file found in `docs/_build/html` directory.
```
